dm/common/common.go

```diff
@@ -49,10 +51,20 @@ func IsErrNetClosing(err error) bool {
 //KeyAdapter used to counstruct etcd key.
 type KeyAdapter interface {
 	Encode(keys ...string) string
-	Decode(key string) []string
+	Decode(key string) ([]string, error)
 	Path() string
 }
 
+func keyAdapterValueLen(s KeyAdapter) int {
```

Comment:
1. Is _key_ length or _value_ length?
2. It's better to put this function as closely as possible to these key adapters.

Reply:
key length

Reply:
But still naming it `***ValueLen`?

Reply:
@lichunzhu ping

Reply:
@csuzhangxc How about `keyAdapterKeysLen`?

Reply:
fine.

Reply:
addressed in https://github.com/pingcap/dm/pull/461/commits/59e36212e713353785fd3c93b9be9e4918c62030

Reply:
addressed in https://github.com/pingcap/dm/pull/461/commits/59e36212e713353785fd3c93b9be9e4918c62030